docs/core/compatibility/aspnet-core/6.0/actionresult-statuscode.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "Breaking change: ActionResult<T> sets StatusCode to 200"
+description: "Learn about the breaking change in ASP.NET Core 6.0 where ActionResult<T> always sets the status code to 200, even if it was set manually."
+ms.date: 05/23/2022
+---
+# ActionResult\<T> sets StatusCode to 200
+
+When returning a `T` in a MVC/API controller action that declares the return type as <xref:Microsoft.AspNetCore.Mvc.ActionResult%601>, the <xref:Microsoft.AspNetCore.Mvc.ObjectResult.StatusCode?displayProperty=nameWithType> is always set to 200, except when the `T` is a <xref:Microsoft.AspNetCore.Mvc.ProblemDetails>.
+
+This change can cause unexpected behavior in some scenarios where you set the status code manually, since previously the <xref:Microsoft.AspNetCore.Mvc.ObjectResult.StatusCode?displayProperty=nameWithType> was `null`. Also, an action filter could be affected by this change if it expects a null value instead of 200.
+
+## Version introduced
+
+ASP.NET Core 6.0
+
+## Previous behavior
+
+Previously, a controller's action that returns `T` and sets `Response.StatusCode` manually generated the expected `202 Accepted` response status code.
+
+```csharp
+// Generates a 202 Accepted response
+public ActionResult<Model> Get()
+{
+    Response.StatusCode = StatusCodes.Status202Accepted;
+    return new Model();
+}
+```
+
+## New behavior
+
+Now, the same controller's action that returns `T` and sets `Response.StatusCode` manually always generates a `200 OK` response.
+
+```csharp
+// Generates a 200 OK response
+public ActionResult<Model> Get()
+{
+    Response.StatusCode = StatusCodes.Status202Accepted;
+    return new Model();
+}
+```
+
+## Type of breaking change
+
+This change can affect [source compatibility](../../categories.md#source-compatibility).
+
+## Reason for change
+
+Returning a status code of `200 OK` is [documented since ASP.NET Core 3.1](/aspnet/core/web-api/action-return-types#actionresultt-type)]. However, it keeps <xref:Microsoft.AspNetCore.Mvc.ObjectResult.StatusCode> as `null` and eventually generates a `200 OK` response only because it's the default. Since the default internal behavior could change, we decided to avoid relying on the default and to explicitly set <xref:Microsoft.AspNetCore.Mvc.ObjectResult.StatusCode> to the expected `200 OK`.
```

Comment:
```suggestion
Returning a status code of `200 OK` is [documented since ASP.NET Core 3.1](/aspnet/core/web-api/action-return-types#actionresultt-type). However, it keeps <xref:Microsoft.AspNetCore.Mvc.ObjectResult.StatusCode> as `null` and eventually generates a `200 OK` response only because it's the default. Since the default internal behavior could change, we decided to avoid relying on the default and to explicitly set <xref:Microsoft.AspNetCore.Mvc.ObjectResult.StatusCode> to the expected `200 OK`.
```